### PR TITLE
arch: add cortex-m0plus support

### PR DESCRIPTION
**Pull request by Joe-Zhucg**
Change-Id: I7e5d13af6574a3d0466e81aa7b548df24f9871b9

### CODE REVIEW

File: cmake/Toolchain/GNU-Baremetal.cmake [hunk 1]
```diff
@@ -19,7 +19,7 @@ foreach(language IN ITEMS ASM C CXX)
             "-mstrict-align -fno-builtin -DAARCH64 -D__ASSEMBLY__ ")
     endif()
 
-    if(CMAKE_SYSTEM_PROCESSOR MATCHES "cortex-m(3|7|33|55)")
+    if(CMAKE_SYSTEM_PROCESSOR MATCHES "cortex-m(0plus|3|7|33|55)")
```
Comment (AhmedImbabyGadallah) [inline cmake/Toolchain/GNU-Baremetal.cmake:22]: is (0| .. missing here ?

Comment (Joe-Zhucg) [reply cmake/Toolchain/GNU-Baremetal.cmake]: updated in new push

File: arch/arm/arm-m/src/arch_nvic.c [hunk 2]
```diff
@@ -264,17 +264,23 @@ static void irq_invalid(void)
 int arch_nvic_init(const struct fwk_arch_interrupt_driver **driver)
 {
     uint32_t ictr_intlinesnum;
+#ifndef ARMV6M
```
Comment (nicola-mazzucato-arm) [inline arch/arm/arm-m/src/arch_nvic.c:267]: I can see a number of ifdef here, which make the code more difficult to read/follow.
Can you please propose a specific arch_nvic_init_v6m() instead?

**Review by nicola-mazzucato-arm — CHANGES_REQUESTED** [review]
Please also add a brief description in the commit message, thanks.

File: arch/arm/arm-m/src/arch_nvic.c (hunk 2 above)
Comment (Joe-Zhucg) [reply arch/arm/arm-m/src/arch_nvic.c]: updated, add additional init function will cause compile error as some instruction used in original init function are not defined for V6M.

**Comment by leandro-arm** [comment]
Jenkins: ok to test

**Comment by leandro-arm** [comment]
Jenkins: ok to test

**Review by leandro-arm — CHANGES_REQUESTED** [review]
Hi! Can I please ask you to rebase your patch?
Thanks

**Comment by Joe-Zhucg** [comment]
Rebased to the latest master branch

**Comment by tarek-arm** [comment]
Jenkins: OK to test

**Comment by leandro-arm** [comment]
@Joe-Zhucg Can you also please update the top banner in all modified files to be
`Copyright <created year> - 2023`. ?
You can check it by running `tools/check_copyright.py`

Thanks!

**Comment by Joe-Zhucg** [comment]
updated copyright

**Comment by leandro-arm** [comment]
Jenkins: ok to test

**Review by leandro-arm — APPROVED** [review]

**Pull request merged**